logs/field.go

```diff
@@ -0,0 +1,260 @@
+package logs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Field represents typed log field (a key-value pair). Adapters should determine
+// Field's type based on Type and use the corresponding getter method to retrieve
+// the value:
+//
+//	switch f.Type() {
+//	case logs.IntType:
+//	    var i int = f.Int()
+//	    // handle int value
+//	case logs.StringType:
+//	    var s string = f.String()
+//	    // handle string value
+//	//...
+//	}
+//
+// Getter methods must not be called on fields with wrong Type (e.g. calling String()
+// on fields with Type != StringType).
+// Field must not be initialized directly as a struct literal.
+type Field struct {
+	ftype FieldType
+	key   string
+
+	vint int64
+	vstr string
+	vany interface{}
+}
+
+func (f Field) Type() FieldType {
+	return f.ftype
+}
+
+func (f Field) Key() string {
+	return f.key
+}
+
+// String is a value getter for fields with StringType type
+func (f Field) String() string {
+	return f.vstr
+}
+
+// Int is a value getter for fields with IntType type
+func (f Field) Int() int {
+	return int(f.vint)
+}
+
+func (f Field) Int64() int64 {
+	return f.vint
+}
+
+// Bool is a value getter for fields with BoolType type
+func (f Field) Bool() bool {
+	return f.vint != 0
+}
+
+// Duration is a value getter for fields with DurationType type
+func (f Field) Duration() time.Duration {
+	return time.Nanosecond * time.Duration(f.vint)
+}
+
+// Strings is a value getter for fields with StringsType type
+func (f Field) Strings() []string {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.([]string)
+}
+
+// Error is a value getter for fields with ErrorType type
+func (f Field) Error() error {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.(error)
+}
+
+// Any is a value getter for fields with AnyType type
+func (f Field) Any() interface{} {
+	return f.vany
+}
+
+// Stringer is a value getteer for fields with StringerType type
+func (f Field) Stringer() fmt.Stringer {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.(fmt.Stringer)
+}
+
+// Fallback returns default string representation of Field value.
+// It should be used by adapters that don't support f.Type directly.
+// Non-nil error indicates some fatal marshaling error and is most
+// likely a sign of invalid Field initialization.
+func (f Field) Fallback() (repr string, err error) {
+	defer func() {
+		if p := recover(); p != nil {
+			if e, ok := p.(error); ok {
+				err = e
+			} else {
+				err = fmt.Errorf("%v", p)
```

Comment:
what if use: "ydb: panic with ..." and wrap the error with xerrors.Wrap + xerrors.WithStacktrace?